### PR TITLE
refactoring

### LINKED ISSUE #122: Refactoring
I think a general refactoring would be quite nice before we release v0.2.0

Some things to consider:
- Creating `latex.jl` and `transitions.jl` to have less inside `Javis.jl`
- Move constructors for `Action` and `SubAction` in a different file as they take a lot of space ...
- [x] Have a general method for getting the current interpolation variable `t` for transformations which makes #42 a lot easier
- Renaming confusion of `Transformation` and `Transition` maybe I need some English advice @TheCedarPrince :smile:

### PR DESCRIPTION
**Pull request by Wikunia**
## PR Checklist

Refactoring PR

**Link to relevant issue(s)**
Closes #122 

**How did you address these issues with this PR? What methods did you use?**
I created a bunch of new files for structs and other functionalities to keep everything easier to find.

### CODE REVIEW

File: src/structs/Action.jl [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        Action(frames_or_id, nothing, func, args...; kwargs...)
+    else
+        Action(:same, frames_or_id, func, args...; kwargs...)
+    end
+end
+
+"""
+    Action(func::Function, args...)
+
+Similar to the above but uses the same as frames as the action above.
```
Comment (TheCedarPrince) [inline src/structs/Action.jl:69]: Typo: "the same as frames as" -> "the same frames as the action above"

File: src/structs/Frames.jl [hunk 2]
```diff
@@ -0,0 +1,58 @@
+"""
+    Frames
+
+Stores the actual computed frames and the user input
+which can be i.e. `:same` or `Rel(10)`.
```
Comment (TheCedarPrince) [inline src/structs/Frames.jl:5]: Remove the "i.e."

**Review by TheCedarPrince — CHANGES_REQUESTED** [review]
Really glad you split things up to make it a bit more manageable and logical rather than having everything in one file. Makes a lot of sense to me.

Looking back through old documentation, I noticed a few typos. Otherwise, I think this is good.

Also, I noticed there are some code coverage issues in `latex.jl`, `structs/SubAction.jl`, `symbol_values.jl` file -- do you want to address this first?

**Comment by Wikunia** [comment]
We have the same coverage as before so I think this is fine or can be addressed later.

**Pull request merged**